mysql/lib/dialect/mysql2_driver_dialect.ts

```diff
@@ -52,4 +52,6 @@ export class MySQL2DriverDialect implements DriverDialect {
   getAwsPoolClient(props: PoolOptions): AwsPoolClient {
     return new AwsMysqlPoolClient(props);
   }
+
+  setKeepAliveProperties(props: Map<string, any>, keepAliveProps: any) {}
```

Comment:
Since mysql driver doesn't support tcpKeepAlive I'd suggest to check keepAliveProp and if it set then raise an exception saying that the driver doesn't support it.